tests/test_dates.py

```diff
@@ -162,4 +164,42 @@ def test_get_dates(input, expected):
     ]
 )
 def test_last_day(year,month,expected):
-    assert last_day(year,month) == expected
\ No newline at end of file
+    assert last_day(year,month) == expected
+
+@pytest.mark.parametrize(
+    "thing,date,expected",
+    [
+        ({'begin_day': 1, 'begin_month': 5, 'begin_year': 1950, 'end_day': 2, 'end_month': 5, 'end_year': 3000}, None, True),
```

Comment:
I don't like the first two tests as it seems to me they are testing magic.    just set a date (doesn't have to be in paramerize, just hardcode a global TEST_DATE at the top of the file, and test everything (``is_current``, ``has_started`` etc) against that date.  Then there is no magic (even if it is mild magic) in your tests.

Actually, why don't you use the suite of tests I already wrote for the ``is_current`` example that I presented in the group meeting.  It tests all the edge cases.  Use the same suite of tests for the has_started etc., but change the logic based on what you expect to happen.

Reply:
Sounds good I will work on that. Do you agree that we can just entirely remove the is_since and is_before methods?